jetty-util/src/main/java/org/eclipse/jetty/util/HostPort.java

```diff
@@ -24,9 +26,97 @@
  */
 public class HostPort
 {
+    private static final Logger LOG = LoggerFactory.getLogger(HostPort.class);
     private final String _host;
     private final int _port;
 
+    public static HostPort unsafe(String authority)
```

Comment:
needs javadoc to say what is unsafe about it.

Reply:
Added